source/extensions/access_loggers/grpc/grpc_access_log_utils.cc

```diff
@@ -300,18 +301,9 @@ void Utility::extractCommonAccessLogProperties(
   }
 
   for (const auto& key : config.filter_state_objects_to_log()) {
-    if (auto state = stream_info.filterState().getDataReadOnlyGeneric(key); state != nullptr) {
-      ProtobufTypes::MessagePtr serialized_proto = state->serializeAsProto();
-      if (serialized_proto != nullptr) {
-        auto& filter_state_objects = *common_access_log.mutable_filter_state_objects();
-        ProtobufWkt::Any& any = filter_state_objects[key];
-        if (dynamic_cast<ProtobufWkt::Any*>(serialized_proto.get()) != nullptr) {
-          any.Swap(dynamic_cast<ProtobufWkt::Any*>(serialized_proto.get()));
-        } else {
-          any.PackFrom(*serialized_proto);
-        }
-      }
-    }
+    extractFilterStateData(stream_info.filterState(), key, common_access_log);
+    extractFilterStateData(*stream_info.upstreamInfo()->upstreamFilterState().get(), key,
```

Comment:
What will happen if this key is exist in both filter state and upstream filter state?

Reply:
keys have direction, https://github.com/istio/proxy/blob/master/extensions/common/context.h#L29-L33, so downstream_peer will not be in upstream side stream info and upstream_peer will not be in downstream side streaminfo

Reply:
This is a common change that not just effect istio proxy. All others that try to log filter state in gRPC logger will be effected.
It possible that some one add some data in both filter state and upstream filter state with same key.

My suggestion would be try the downstream filter state first. And if no any data found, try the upstream filter state again.

Reply:
sgtm|

Reply:
See the Wasm related code for this. Makes sense to do what @wbpcode recommended, as that is what Wasm does.

Reply:
Thanks @kyessenov and @wbpcode . I have updated the changes as per the suggestion. Please see if looks ok now.
Working to add tests in test/extensions/access_loggers/grpc/tcp_grpc_access_log_integration_test.cc